lib/puppetserver/ca/action/generate_csr.rb

```diff
@@ -0,0 +1,123 @@
+require 'optparse'
+
+require 'puppetserver/ca/config/puppet'
+require 'puppetserver/ca/errors'
+require 'puppetserver/ca/local_certificate_authority'
+require 'puppetserver/ca/utils/cli_parsing'
+require 'puppetserver/ca/utils/file_system'
+require 'puppetserver/ca/utils/signing_digest'
+
+module Puppetserver
+  module Ca
+    module Action
+      class GenerateCsr
+        include Puppetserver::Ca::Utils
+
+        SUMMARY = 'Generate a CSR for an intermediate signing CA cert'
+        BANNER = <<-BANNER
+Usage:
+  puppetserver ca generate-csr [--help]
+  puppetserver ca generate-csr --output-dir PATH [--config PATH] [--ca-name NAME]
+
+Description:
+  Setup a root and intermediate signing CA for Puppet Server
+  and store generated CA keys, certs, crls, and associated
+  master related files on disk.
```

Comment:
I don't think this is the correct description

Reply:
fixed